android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.util;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.text.TextUtils;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.Priority;
+import com.bumptech.glide.RequestBuilder;
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+import com.bumptech.glide.load.resource.bitmap.CenterInside;
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
+import com.bumptech.glide.load.resource.gif.GifDrawable;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ImageLoader {
+    private static final List<String> ANIMATED_lIST = Arrays.asList(".gif");
+
+    public static void loadNft(
+            String url, ImageView imageView, Context context, boolean isCircular) {
+        if (isAnimated(url)) {
+            loadGif(url, imageView, context, isCircular);
+        } else {
+            loadImg(url, imageView, context, isCircular);
+        }
+    }
+
+    public static void loadImg(
+            String url, ImageView imageView, Context context, boolean isCircular) {
+        RequestBuilder<Drawable> request =
+                Glide.with(context).load(url).transform(new CenterInside(), new RoundedCorners(24));
+        if (isCircular) {
+            request = request.circleCrop();
+        }
+        request.priority(Priority.IMMEDIATE)
+                .diskCacheStrategy(DiskCacheStrategy.ALL)
+                .into(imageView);
+    }
+
+    public static void loadGif(
+            String url, ImageView imageView, Context context, boolean isCircular) {
+        RequestBuilder<GifDrawable> request = Glide.with(context).asGif().load(url).transform(
+                new CenterInside(), new RoundedCorners(24));
```

Comment:
Wouldn't be better to use `CenterFit` instead of `CenterInside`? 
`CenterFit` it's similar to `CenterInside` but will also cover the case where the image is much smaller of the view, and it will increase the dimension keeping the same ratio.

Reply:
You mean `FitCenter`. I think it's fine to show smaller images as it is than stretching to avoid pixelation.
Probably your task is more suitable to check which looks better for small images on NFTDetails screen.
BTW, I think we can do these possible updates or modifications later when required?

Reply:
Yes, absolutely 👍